src/pybind/mgr/volumes/fs/subvolume.py

```diff
@@ -189,6 +189,29 @@ def get_subvolume_path(self, spec):
             raise VolumeException(-e.args[0], e.args[1])
         return path
 
+    def get_dir_names(self, path):
+        """
+        Get the directory names in a given root path
+        :param path: the root path
+        :return: the list of directory names in json format
+        """
+        try:
+            dir_handle = self.fs.opendir(path)
+        except cephfs.ObjectNotFound:
+            raise VolumeException(-errno.ENOENT,
+                                  "Path '{0}' not found".format(path))
+        except cephfs.Error as e:
+            raise VolumeException(-e.args[0], e.args[1])
+
+        dirs = []
+        d = self.fs.readdir(dir_handle)
+        while d:
+            if d.d_name not in (b".", b".."):
+                dirs.append({'name': d.d_name})
```

Comment:
d.d_name is  `bytes` object in py3. You need to convert it to string type to JSON serialize it, else you'll hit TypeError in py3 environment.
```
d.d_name.decode('utf-8')
```